File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/mst.cs
/**
 * C# Implementation: Minimum Spanning Tree (Kruskal & Prim)
 * Invariant: O(E log V) with Disjoint Set Union
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class MinimumSpanningTreeKruskalPrim<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/java/mst.java
/**
 * Java Implementation: Minimum Spanning Tree (Kruskal & Prim)
 * Invariant: O(E log V) with Disjoint Set Union
 */
package seed.dsa.core;

import java.util.*;

public class MinimumSpanningTreeKruskalPrim<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
